Content.Server/Radiation/Systems/GeigerSystem.cs

```diff
@@ -0,0 +1,187 @@
+        SubscribeLocalEvent<GeigerComponent, GotUnequippedHandEvent>(OnUnequippedHand);
+
+        SubscribeLocalEvent<OnRadiationSystemUpdated>(OnUpdate);
+        SubscribeLocalEvent<GeigerComponent, ComponentGetState>(OnGetState);
+    }
+
+    private void OnActivate(EntityUid uid, GeigerComponent component, ActivateInWorldEvent args)
+    {
+        if (args.Handled || component.AttachedToSuit)
+            return;
+        args.Handled = true;
+
+        SetEnabled(uid, component, !component.IsEnabled);
+    }
+
+    private void OnEquipped(EntityUid uid, GeigerComponent component, GotEquippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, true);
+        SetUser(uid, component, args.Equipee);
+    }
+
+    private void OnEquippedHand(EntityUid uid, GeigerComponent component, GotEquippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, args.User);
+    }
+
+    private void OnUnequipped(EntityUid uid, GeigerComponent component, GotUnequippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, false);
+        SetUser(uid, component, null);
+    }
+
+    private void OnUnequippedHand(EntityUid uid, GeigerComponent component, GotUnequippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, null);
+    }
+
+    private void OnUpdate(OnRadiationSystemUpdated ev)
+    {
+        var query = EntityQuery<GeigerComponent, RadiationReceiverComponent>();
+        foreach (var (geiger, receiver) in query)
+        {
+            var rads = receiver.CurrentRadiation;
+            SetCurrentRadiation(geiger.Owner, geiger, rads);
+        }
+    }
+
+    private void OnGetState(EntityUid uid, GeigerComponent component, ref ComponentGetState args)
+    {
+        args.State = new GeigerComponentState
+        {
+            CurrentRadiation = component.CurrentRadiation,
+            DangerLevel = component.DangerLevel,
+            IsEnabled = component.IsEnabled
+        };
+    }
+
+    private void SetCurrentRadiation(EntityUid uid, GeigerComponent component, float rads)
+    {
+        // check that it's approx equal
+        if (MathHelper.CloseTo(component.CurrentRadiation, rads, ApproxEqual))
+            return;
+
+        var curLevel = component.DangerLevel;
+        var newLevel = RadsToLevel(rads);
+
+        component.CurrentRadiation = rads;
+        component.DangerLevel = newLevel;
+
+        if (curLevel != newLevel)
+        {
+            UpdateAppearance(uid, component);
+            UpdateGeigerSound(uid, component);
+        }
+
+        Dirty(component);
+    }
+
+    private void SetUser(EntityUid uid, GeigerComponent component, EntityUid? user)
+    {
+        if (component.User == user)
+            return;
+
+        component.User = user;
+        UpdateGeigerSound(uid, component);
+    }
```

Comment:
See above, server shouldn't care about tracking this and most of this could just be in shared.

Reply:
I tried to control it client-side, but this is really annoying to do. Radiation level is updated on server side, so it's very convenient to set all visual/sound/component state there. 

For some logic I can't use handle state. For example appearance system will ignore any changes during handle state on a client.

Reply:
Doesn't the client just need to know about the rad level?

Instead now the server has to send all of the visual / sound data over to the client instead of just the radiation level.

Reply:
From Discord discussion.

<img width="895" alt="Снимок экрана 2022-11-09 в 19 02 21" src="https://user-images.githubusercontent.com/6161335/200905948-75a880a7-778f-4949-bb2b-6bff946e1ee9.png">

Reply:
I moved current user to uid to shared and synced it with client. Also added some extra logic to check attached entity. Seems to work ok.